apps/openmw/mwworld/worldimp.cpp

```diff
@@ -2681,18 +2684,30 @@ namespace MWWorld
         if (result2.mHit)
             dist2 = (origin - result2.mHitPointWorld).length();
 
-        if (dist1 <= dist2 && result1.mHit)
+        if (result1.mHit)
         {
             target = result1.mHitObject;
             hitPosition = result1.mHitPos;
             if (dist1 > getMaxActivationDistance() && !target.isEmpty() && (target.getClass().isActor() || !target.getClass().canBeActivated(target)))
                 target = NULL;
+            if (!Settings::Manager::getBool("allow AI to hit unintended targets", "Game"))
+            {
+                // Don't let AI actors hit unintended targets. If an unintended target is hit, continue casting the ray past them
+                while (actor != MWMechanics::getPlayer() && !target.isEmpty() && target.getClass().isActor() && !stats.getAiSequence().isInCombat(target))
+                {
+                    result1 = mPhysics->castRay(result1.mHitPos, dest, target, MWPhysics::CollisionType_Actor);
+                    if (result1.mHit)
+                        target = result1.mHitObject;
+                    else
+                        target = NULL;
```

Comment:
I wouldn't rely on a cast from the hit point to work. How do you know you're not just going to hit the same shape again? Worst case this would be an infinite loop.

Instead, I would use a single cast and filter unintended targets via the ResultCallback.


I'd remove the setting for now; really no reason to enable at the moment with the AI being as bad as it is.